### PR TITLE
bug 1412891: Enable brotli on MDN

### PR DESCRIPTION
**Pull request by MatonAnthony**
The brotli compression algorithm provides a more efficient compression algorithm than gzip and as such should be available on MDN.

### CODE REVIEW

**Comment by MatonAnthony** [comment]
I will implement the requested changes.

I thought I just had to apply those commit messages rules on the final rebase. Isn't that the case?
I will pay attention to my branches name in the future.

File: kuma/core/middleware.py [hunk 1]
```diff
@@ -86,6 +89,7 @@ class Forbidden403Middleware(object):
     """
     Renders a 403.html page if response.status_code == 403.
     """
+
```
[the remark this answers is not in the document]
Comment (MatonAnthony) [reply kuma/core/middleware.py]: Yes, it's Pycharm who added it.

**Comment by MatonAnthony** [comment]
I believe it's ready for a second review.

**Comment by jwhitlock** [comment]
@MatonAnthony, thanks for the style updates. 

Thanks for researching the Google ``brotli`` vs ``brotlpy`` issue, and finding https://github.com/evansd/whitenoise/issues/154.  I agree Google's version is the better one going forward. Once we've proved it works, I can submit a PR to update the whitenoise docs.

I spent some time getting my branch [docker-ssl-1218563](https://github.com/jwhitlock/kuma/tree/docker-ssl-1218563) up-to-date, and tested this code with SSL enabled. I was able to get Brotli-compressed content in the browser for the HTML page, but still not for the static assets. I verified that Brotli is enabled in whitenoise, but I wasn't able to get anything other than GZip compression out of the development environment. It appears to just use ``brotli.compress``, so the API is the same.  I'm 50% sure this will work in production, and 90% sure it won't break. We'll do manual testing in staging. I have work to do to understand our static assets.

I added some commits to fix some nits. We don't use PEP8's 80-character lines (yet), but some of these were out of control. Some of the comments were misleading as well - the linked issue says that Google's ``brotli`` _does_ support streaming content, and the response is returned uncompressed if compression results in a longer string or the same length. It seemed easier to add a commit than to go through another round of reviews.

I'll merge when the tests pass, and we'll see if static assets are compressed with Brotli on staging.

**Comment by MatonAnthony** [comment]
Ok, thanks for the extra commits.

I will pay attention to my line length next time.

**Comment by jwhitlock** [comment]
Pushed to stage, and I'm getting some Brotli compression:

screenshot 2018-03-07 10 40 08

Assets are not getting Brotli compression on staging. It is hard to see if this is due to a configuration issue (the assumed issue on development) or Cloudfront, which has interesting behavior around compression - https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/ServingCompressedFiles.html

It appears to work in the browser, for HTML pages, so I think we're OK going to production. I don't think I'll bother filing a bug for the static assets.

**Comment by cristiangraz** [comment]
@jwhitlock CloudFront now supports Brotli compression at the edge or the origin. 
 https://aws.amazon.com/about-aws/whats-new/2020/09/cloudfront-brotli-compression/

**Comment by escattone** [comment]
> @jwhitlock CloudFront now supports Brotli compression at the edge or the origin.
> https://aws.amazon.com/about-aws/whats-new/2020/09/cloudfront-brotli-compression/

@cristiangraz I didn't realize it was already available, so thanks for letting us know! I just enabled it on our development CDN and I suspect you'll see it in production sometime before the end of year.

**Pull request merged**